core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -498,7 +499,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     // map. If an attempt has been updated, it replaces the old attempt in the list.
     val newAppMap = new mutable.HashMap[String, FsApplicationHistoryInfo]()
 
-    applications.synchronized {
+    applicationsLock.synchronized {
```

Comment:
Why not synchronize on `this` instead?

```scala
synchronized {
    // ...
}
```

Reply:
I wanted to really emphasis the purpose of this object used as a monitor to lock the access that would change applications reference. 
Technically, it will work the same if I synchronise on `this`

Do you still prefer to synchronise on `this` ? Completely fine for me to change the code

Reply:
I'm fine with an extra object, was just curious what your reasoning was.